source/src_pw/hamilt_pw_hip.h

```diff
@@ -0,0 +1,175 @@
+#ifndef HAMILT_PW_HIP_H
+#define HAMILT_PW_HIP_H
+
+#include "tools.h"
+
+#ifdef __ROCM
+#include "hipfft.h"
+#include "hipblas.h"
+// #include "rocsolver.h"
+#endif
+
+class Hamilt_PW
+{
+
+public:
+
+    Hamilt_PW();
+    ~Hamilt_PW();
+
+    static int moved;
+
+	void allocate(
+		const int &npwx, // number of plane wave (max)
+		const int &npol, // polarization
+		const int &nkb,  // number of non-local pseudopotential projectors
+		const int &nrxx); // number of grids on this processor
+
+    void cal_err
+    (
+        const int &npw,
+        ModuleBase::ComplexMatrix &psi,
+        const int &nband,
+        double *em,
+        double *err
+    );
+
+	void init_k(const int ik);
+
+	friend class Diago_David;
+	// friend class Diago_CG;
+    template<class T, class T2, class T3> friend class Diago_CG_CUDA;
+	friend class Exx_Lip;
+	friend class Hamilt;
+    friend class Stochastic_Iter;
+
+	void diagH_subspace(const int ik,
+                  const int nstart,
+                  const int nbnd,
+                  const ModuleBase::ComplexMatrix &psi,
+                  ModuleBase::ComplexMatrix &evc,
+                  double *en);
+    
+    void diagH_subspace_cuda(
+        const int ik,
+        const int nstart,
+        const int n_band,
+        const hipblasDoubleComplex *psi_c, // matrix
+        hipblasDoubleComplex *evc, // matrix
+        double *en,
```

Comment:
Please add some annotation for functions in this file.
And be careful for the code indentation format.